intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R

```diff
@@ -0,0 +1,88 @@
+# Set up gene level statistics table for intro to R module
+
+# C. Savonen for ALSF - CCDL
+
+# 2020
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Establish R dir
+intro_r_dir <- file.path(root_dir, "intro-to-R-tidyverse")
+
+# Establish input and output directories
+data_dir <- file.path(intro_r_dir, "data")
+plots_dir <- file.path(output_dir, "plots")
+
+# Create these output directories if they don't exist
+if (!dir.exists(output_dir)) {
+  dir.create(output_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Install limma if not installed
+if (!("limma" %in% installed.packages())) {
+  # Install limma
+  BiocManager::install("limma", update = FALSE)
+}
+
+# Install the Human annotation package
+if (!("org.Hs.eg.db" %in% installed.packages())) {
+  # Install org.Hs.eg.db
+  BiocManager::install("org.Hs.eg.db", update = FALSE)
+}
+
+# Read in the dataset
+gene_df <- readr::read_tsv(file.path(input_dir, 
+                                     "GSE44971.tsv")) %>% 
+  tibble::column_to_rownames("Gene")
+
+# Read in metadata
+metadata <- readr::read_tsv(file.path(input_dir, 
+                                      "metadata_GSE44971.tsv")) %>%
+  dplyr::select(-which(apply(is.na(.), 2, all))) %>% 
+  dplyr::mutate(tissue = as.factor(tissue))
+
+## Put samples' data and metadata in the same order
+# Make the data in the order of the metadata
+gene_df <- gene_df %>% 
+  dplyr::select(metadata$geo_accession)
+
+# Create the design matrix from the genotype information
+des_mat <- model.matrix(~metadata$tissue)
```

Comment:
What happens if we make this a bit more complex? I can't remember what limma output will look like, but why not tissue* sex for a simple factorial design? Sample size in this data set may get a bit small, but from a quick glance, it appears there are 3 in the smallest group (Female normal). And we all know 3 is the correct number.

If this works out, you may have to add a bit more cleaning to *keep it tidy™*.

Reply:
Okay. I'm in a bit of a `model.matrix` and `limma` rabbit hole. What is our main question(s) and contrasts we want to make? Is this correct? 
1) Differences between normal and tumor
2) Differences between sexes
3) Interaction of sex and tumor?

Reply:
After playing around with this, I feel like doing a 2x2 design makes this more complicated than needed when this module is mainly focused on getting the participants to feel comfortable with data.frames and ggplot2. I don't necessarily want to add an added layer of complexity that seems to not add toward the main learning objective. What do you think @jashapiro ?

Reply:
That would be fine. Might as well do it correctly, which would have all three of those.
So you ought to be able to just do: 
```
des_mat <- model.matrix(~metadata$tissue * metadata$refinebio_sex)
```
to get all of those in the model.

Reply:
^I mean yes, however if you want the logFC for each comparison, `toptable` will not automatically do that. And then I went down a rabbit hole of what comparisons do we care about.

Reply:
I just feel like its very easy to get distracted by this 2x2 experimental model and I am not sure (at this point in time) how this aids the participant's learning of tidyverse. Maybe I need a clearer idea of your vision behind doing a 2x2 model.

Reply:
A fair point. My vision is actually pretty simple here. I just want more than a single comparison, so we can filter results down to just one comparison for plotting at first, then show them how we can make multiples with facet_grid. 

The idea is to give little hints at the power available. Not go down rabbit holes.

Reply:
Alright. I tried to follow 8.7 instead. The contrasts probably need an extra pair of eyes to check.